specification/flag-evaluation/flag-evaluation.md

```diff
@@ -165,68 +165,66 @@ FlagEvaluationDetails<MyStruct> myStructDetails = client.getObjectDetails<MyStru
 
 ```
 
-##### Requirement 1.10
+##### Requirement 1.4.2
 
 > The `evaluation details` structure's `value` field **MUST** contain the
 > evaluated flag value.
 
-##### Condition 1.11
+##### Condition 1.4.3
 
 > The language supports generics (or an equivalent feature).
 
-###### Conditional Requirement 1.11.1
+###### Conditional Requirement 1.4.3.1
 
 > The `evaluation details` structure **SHOULD** accept a generic argument (or
 > use an equivalent language feature) which indicates the type of the wrapped
 > `value` field.
 
-##### Requirement 1.12
+##### Requirement 1.4.4
 
 > The `evaluation details` structure's `flag key` field **MUST** contain the
 > `flag key` argument passed to the detailed flag evaluation method.
 
-##### Requirement 1.13
+##### Requirement 1.4.5
 
 > In cases of normal execution, the `evaluation details` structure's `variant`
 > field **MUST** contain the value of the `variant` field in the
 > `flag resolution` structure returned by the configured `provider`, if the
 > field is set.
 
-##### Requirement 1.14
+##### Requirement 1.4.6
 
 > In cases of normal execution, the `evaluation details` structure's `reason`
 > field **MUST** contain the value of the `reason` field in the
 > `flag resolution` structure returned by the configured `provider`, if the
 > field is set.
 
-##### Requirement 1.15
+##### Requirement 1.4.7
 
 > In cases of abnormal execution, the `evaluation details` structure's
 > `error code` field **MUST** identify an error occurred during flag evaluation,
 > having possible values `"PROVIDER_NOT_READY"`, `"FLAG_NOT_FOUND"`,
 > `"PARSE_ERROR"`, `"TYPE_MISMATCH"`, or `"GENERAL"`.
 
-##### Requirement 1.16
+##### Requirement 1.4.8
 
 > In cases of abnormal execution (network failure, unhandled error, etc) the
 > `reason` field in the `evaluation details` **SHOULD** indicate an error.
 
-##### Requirement 1.17
-
-> The `evaluation options` structure's `hooks` field denotes a collection of
-> hooks that the client **MUST** execute for the respective flag evaluation, in
-> addition to those already configured.
-
-See [hooks](./hooks.md) for details.
-
-##### Requirement 1.18
+##### Requirement 1.4.9
 
 > Methods, functions, or operations on the client **MUST NOT** throw exceptions,
 > or otherwise abnormally terminate. Flag evaluation calls must always return
 > the `default value` in the event of abnormal execution. Exceptions include
-> functions or methods for the purposes for configuration or setup.
+> functions or methods for the purposes for configuration or setup, and
+> user-defined code such as hooks.
 
-##### Requirement 1.19
+User-defined code in `error` and `finally` hooks may throw or otherwise
+abnormally terminate. This has the benefit of providing a means of making errors
+optionally "loud" for debugging purposes (errors can be re-thrown in `error`
+hooks, for instance).
+
```

Comment:
This is already the case for `finally`, I think it makes sense for `error` too...